keps/sig-storage/2268-non-graceful-shutdown/README.md

```diff
@@ -0,0 +1,466 @@
+* Feature is deployed in production and have gone through at least one Kubernetes upgrade.
+* More rigorous forms of testing—e.g., downgrade tests and scalability tests.
+* Allowing time for feedback.
+
+### Upgrade / Downgrade Strategy
+
+once enabled by default, the upgrades should be smooth as the default behaviour is off by default. when downgrading/rollbacking with this behaviour enabled, there should be no issues as the control plane with the logic is still there.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [ ] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: NonGracefulFailover
+    - Components depending on the feature gate: kube-controller-manager, kubelet
+  - [ ] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+      Yes.
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+      No.
+
+* **Does enabling the feature change any default behavior?**
+  <!--
+  Any change of default behavior may be surprising to users or break existing
+  automations, so be extremely careful here.
+  -->
+  Yes. If this feature is enabled, the pod gc controller in kube-controller-manager
+  would need to ensure that `SafeDetach` option in `CSIDriver` is true before
+  forcefully deleting pods.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+  Describe the consequences on existing workloads (e.g., if this is a runtime
+  feature, can it break the existing applications?).
+  Yes. If this feature is disable once it has been enabled, it falls back to
```

Comment:
What happens if this feature is disabled part way the process?  For instance, if the node has been tainted, will the automatic untainting happen or will it be stuck?

It's ok for some aspects to require manual interaction, but we need to enumerate what is required.

Reply:
The current design says no automatic taint-removing feature.